### PR TITLE
filter update

### PR DESCRIPTION
**Pull request by westwoodworm**
as branch name said

### CODE REVIEW

File: src/Pachyderm/Orm/Model.php [hunk 1]
```diff
@@ -348,11 +348,18 @@ public static function pagination(array $params): SQLBuilder
         unset($params['order']);
 
         if (!empty($params['filter'])) {
-            $query = new QueryBuilder($params['filter']);
+            $conditions = json_decode($params['filter']);
```
Comment (mermetbt) [inline src/Pachyderm/Orm/Model.php:351]: Why is it decoding json?

File: src/Pachyderm/Orm/Model.php [hunk 2]
```diff
@@ -348,11 +348,18 @@ public static function pagination(array $params): SQLBuilder
...
+            $query = new QueryBuilder();
+            foreach ($conditions as $condition) {
```
Comment (mermetbt) [inline src/Pachyderm/Orm/Model.php:353]: This makes a different behavior of what was done before.

**Review by mermetbt — CHANGES_REQUESTED** [review]

File: src/Pachyderm/Orm/Model.php (hunk 1 above)
Comment (westwoodworm) [reply src/Pachyderm/Orm/Model.php:351]: The filter param from front is in json encoded, here it parses it into array

File: src/Pachyderm/Orm/Model.php (hunk 2 above)
Comment (westwoodworm) [reply src/Pachyderm/Orm/Model.php:353]: The previous pass filter seems doesn't work and not used anywhere else. The 'filter' here is a special reserved key for those kind of queries.

File: src/Pachyderm/Orm/Model.php (hunk 1 above)
Comment (mermetbt) [reply src/Pachyderm/Orm/Model.php:351]: I can understand the code, I cannot understand the reason.

File: src/Pachyderm/Orm/Model.php (hunk 2 above)
Comment (mermetbt) [reply src/Pachyderm/Orm/Model.php:353]: This doesn't work as it was before. You've destroyed a working feature on a framework used by many apps.

**Comment by mermetbt** [comment]
This pull request will never be merged. Code is destroying working features and doesn't improve any part of the framework.

**Pull request closed without merging**